src/lj_str_hash.c

```diff
@@ -5,39 +5,40 @@
  * to 128 bytes of given string.
  */
 
-#ifndef _LJ_STR_HASH_X64_H_
-#define _LJ_STR_HASH_X64_H_
+#define lj_str_hash_c
+#define LUA_CORE
 
-#if defined(__SSE4_2__) && defined(__x86_64) && defined(__GNUC__)
+#include "lj_str_hash.h"
+
+#if LJ_OR_STRHASHCRC32
 
-#include <stdint.h>
 #include <sys/types.h>
 #include <unistd.h>
 #include <time.h>
 #include <smmintrin.h>
+#include "lj_vm.h"
 
-#include "../../lj_def.h"
-
-#undef LJ_AINLINE
-#define LJ_AINLINE
+#ifndef F_CPU_SSE4_2
+#define F_CPU_SSE4_2 	(1 << 20)
+#endif
 
 #ifdef __MINGW32__
-#define random()  ((long) rand())
-#define srandom(seed)  srand(seed)
+#define random() 	((long) rand())
+#define srandom(seed) 	srand(seed)
 #endif
 
-static const uint64_t* cast_uint64p(const char* str)
+static LJ_AINLINE const uint64_t* cast_uint64p(const char* str)
 {
   return (const uint64_t*)(void*)str;
 }
 
-static const uint32_t* cast_uint32p(const char* str)
+static LJ_AINLINE const uint32_t* cast_uint32p(const char* str)
 {
   return (const uint32_t*)(void*)str;
 }
 
 /* hash string with len in [1, 4) */
-static LJ_AINLINE uint32_t lj_str_hash_1_4(const char* str, uint32_t len)
+static LJ_NOINLINE uint32_t lj_str_hash_1_4(const char* str, uint32_t len)
```

Comment:
My previous comment somehow is gone. I  try to understand this change again. How come you don't want to have this function inlined? As far as I can tell, it's perfect candidate for inlining.

Reply:
Because previous versions of this patch have a particular treatment for `LJ_AINLINE`: https://github.com/openresty/luajit2/blob/4589430e611042980368f5f1ea0ac4114def6510/src/x64/src/lj_str_hash_x64.h#L21-L22

I agree that it seems like a good candidate for inlining, but my reasoning was to wait until we have some micro benchmarks to run before making this change.